tt_metal/impl/device/device.cpp

```diff
@@ -2322,6 +2328,23 @@ std::shared_ptr<TraceBuffer> Device::get_trace(const uint32_t tid) {
     }
 }
 
+DeviceAddr Device::get_dev_addr(CoreCoord phys_core, HalMemAddrType addr_type) {
```

Comment:
besides `Device::core_type_from_physical_core(const CoreCoord &physical_core)` and logical <-> physical APIs most of our APIs accept logical coords. any reason this needs/is better with physical coords?

Reply:
I don't have a great answer to this: but most of the code that needs the memory map is already dealing w/ physical coords at the time of the call (watcher, dprint, etc), so I went this way.  I think this makes sense for now, though can believe that we'd want to change it at some point